diff --git a/src/request.c b/src/request.c
index 63174d0..090da42 100644
--- a/src/request.c
+++ b/src/request.c
@@ -44,7 +44,7 @@
 #endif

 #define USER_AGENT_SIZE 256
-#define REQUEST_STACK_SIZE 32
+#define REQUEST_STACK_SIZE 256
 #define SIGNATURE_SCOPE_SIZE 64

 //#define SIGNATURE_DEBUG
@@ -1223,12 +1223,14 @@ static S3Status setup_curl(Request *request,
         snprintf(header, sizeof(header), "Content-Length: %llu",
                  (unsigned long long) params->toS3CallbackTotalSize);
         request->headers = curl_slist_append(request->headers, header);
+        request->headers = curl_slist_append(request->headers, "Expect:");
         request->headers = curl_slist_append(request->headers,
                                              "Transfer-Encoding:");
     }
     else if (params->httpRequestType == HttpRequestTypeCOPY) {
         request->headers = curl_slist_append(request->headers,
                                              "Transfer-Encoding:");
+        request->headers = curl_slist_append(request->headers, "Expect:");
     }

     append_standard_header(hostHeader);

